### PR TITLE
Typefully integration

### PR DESCRIPTION
**Pull request by thomaswang**
👋 @linuz90, @frankdilo

### CODE REVIEW

**Comment by thomaswang** [comment]
CleanShot 2022-01-22 at 22 28 30@2x

File: extension/content/main.js [hunk 1]
```diff
@@ -17,30 +17,80 @@ const addStyles = (id, css) => {
...
+    const typefullyLink = document.createElement("a");
+    typefullyLink.id = "typefully-link";
+    typefullyLink.href = "https://typefully.com";
+    typefullyLink.className = "typefully";
+
+    const typefullyText = document.createElement("span");
+    typefullyText.innerText = "Continue Thread in Typefully";
+
+    const typefullyMark = document.createElement("span");
+    typefullyMark.innerHTML = "";
```
Comment (thomaswang) [inline extension/content/main.js:68]: @linuz90 for the Typefully svg

**Comment by thomaswang** [comment]
image

Copying over screenshot from @linuz90

**Comment by thomaswang** [comment]
CleanShot 2022-01-27 at 19 14 44@2x

@linuz90 the background color for the button is always white, we could potentially get the background color of the modal and apply that to the button instead?

**Comment by linuz90** [comment]
Oh right, totally, that would be even better!

**Comment by thomaswang** [comment]
CleanShot 2022-02-02 at 10 27 37@2x

@linuz90 got the computed style from the modal! Removed the hover style, tried using a `filter: brightness(0.9)` style but it affects the text too. Not sure if there's a better hover style that we could apply here.

**Comment by thomaswang** [comment]
CleanShot 2022-02-09 at 20 05 13

**Comment by thomaswang** [comment]
CleanShot 2022-02-09 at 21 08 25

@frankdilo my initial approach is to get `div`'s with `tweetTextarea` and concat the strings together, but it looks like something as basic as links would break under this condition since they are contained with a separate `span`

**Comment by thomaswang** [comment]
CleanShot 2022-02-09 at 21 12 45

Alternatively, just use "Write thread in Typefully"

**Comment by thomaswang** [comment]
Okay I did more investigating and it looks like I can iterate through `tweetTextarea` and then iterate through each node's `span` with `data-text="true"` and concat with newline

```js
document.querySelectorAll('[data-testid="tweetTextarea_0"] [data-text="true"]').forEach((item) => {console.log(item)})
```

**Pull request merged**